source/node_test.go

```diff
@@ -361,13 +374,21 @@ func testNodeSourceEndpoints(t *testing.T) {
 			_, err := kubernetes.CoreV1().Nodes().Create(context.Background(), node, metav1.CreateOptions{})
 			require.NoError(t, err)
 
+			if tc.exposeInternalIPv6 == nil {
+				tc.exposeInternalIPv6 = new(bool)
+				*tc.exposeInternalIPv6 = true
+			}
+
+			fmt.Printf("node: %v %v\n", tc.nodeName, *tc.exposeInternalIPv6)
+
 			// Create our object under test and get the endpoints.
 			client, err := NewNodeSource(
 				context.TODO(),
 				kubernetes,
 				tc.annotationFilter,
 				tc.fqdnTemplate,
 				labelSelector,
+				*tc.exposeInternalIPv6,
```

Comment:
why it's a reference? Could we have it just a primitive bool with true or false?

Reply:
we could make it a primitive but since the flag is set to be default as true for now I thought we could use a pointer and if its nil then assign it as true else manually set it to false.. otherwise for each test I would have to add an entry with the value as true

Reply:
Can't see a reason how this could be accepted

Reply:
Do you want to replace it with a primitive instead?

Reply:
Indeed